web3js-ext/test/role_based_key_feepayer.spec.ts

```diff
@@ -0,0 +1,140 @@
+        roleTransactionAccount = generateTemporaryAccount();
+        roleAccountUpdate = generateTemporaryAccount();
+        roleFeePayerAccount = generateTemporaryAccount();
+
+        // Feedback 5. PubKey name change
+        const roleTransactionAccountPubkey = getPublicKeyFromPrivate(roleTransactionAccount.privateKey);
+        const roleAccountUpdatePubKey = getPublicKeyFromPrivate(roleAccountUpdate.privateKey);
+        const roleFeePayerPubkey = getPublicKeyFromPrivate(roleFeePayerAccount.privateKey);
+
+        const updateTx = {
+            type: TxType.AccountUpdate,
+            from: roleAccountUpdate.address,
+            gasLimit: 100000,
+            key: {
+                type: AccountKeyType.RoleBased,
+                keys: [
+                    { type: AccountKeyType.Public, key: roleTransactionAccountPubkey },
+                    { type: AccountKeyType.Public, key: roleAccountUpdatePubKey },
+                    { type: AccountKeyType.Public, key: roleFeePayerPubkey }
+                ]
+            }
+        };
+
+        const signedUpdateTx = await roleAccountUpdate.signTransaction(updateTx);
+        assert.isNotNull(signedUpdateTx.transactionHash, "Account update transaction should succeed");
+    });
+
+    // Test Case 1: Sending a normal transaction with RoleTransaction key
+    it("1. Sending a normal transaction with RoleTransaction key", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        const signedTx = await roleTransactionAccount.signTransaction(valueTx);
+        assert.isNotNull(signedTx.transactionHash, "RoleTransaction transaction should succeed");
+    });
+
+    // Test Case 2: Attempting to sign a regular transaction with RoleFeePayer key (should fail)
+    it("2. Attempting to sign a regular transaction with RoleFeePayer key (failure test)", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        try {
+            const signedTx = await roleFeePayerAccount.signTransaction(valueTx);
+            assert.fail("RoleFeePayer key should not sign a ValueTransfer transaction.");
+        } catch (error: any) {
+            assert.isTrue(true, "Error occurred as expected due to role mismatch");
+        }
+    });
+
+    // Test Case 3: Fee Delegated transaction signed by RoleFeePayer key (should succeed)
+    it("3. Fee Delegated transaction signed by RoleTransaction and RoleFeePayer keys", async function () {
+        // Feedback 4. Create a new user account (different user than FeePayer)
+        const userAccount = generateTemporaryAccount();
```

Comment:
Hello @kjeom, I have reflected your feedback as follows.

[Feedback 4]
- As you mentioned, I have supplemented it so that a general user(userAccount) signs the transaction first, and then an additional signature is made with roleFeePayerAccount so that FeePayer can pay the fee on behalf of the general user.

[Feedback 5]
- minor, but changed the pubKey name you told.

[Feedback 6]
- Except for checking for errors, all other console.logs have been removed.

[New Error]
- I changed it as you advised, but I am facing a new error called "insufficient funds for transfer" which is different from the previous RLP error.
![image](https://github.com/user-attachments/assets/f535584e-69cc-4d55-87d1-d9c226b25c7c)

[Questions]
So, I have 2 questions about this.
1. Are TxType and Role-based key related to each other? (For Example, is it possible to sign Tx.Type.AccountUpdate only with roleAccountUpdatePubKey? or sign Tx.Type.FeeDelegatedValueTransferonly only with roleFeePayerPubkey?)

2. Is it true that the error "insufficient funds for transfer" is caused by the fact that the newly created account does not have kaia(fund)?

Reply:
@minminkikiki 
1. yes accountUpdate only can be signed by roleAccountUpdatePrivateKey, not Pubkey when the account is role-based. And if you want to pay the gasFee as a feePayer with role-based account, yes you should sign it with roleFeePayerPrivateKey.
2. No, I think you should fill all kinds of field manually. This error occurred when calling some functions like estimateGas, getting current nonce. I think we need to add all fields manually to avoid calling the RPC functions to the node.